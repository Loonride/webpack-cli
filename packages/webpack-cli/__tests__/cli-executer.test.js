jest.mock('../lib/runner');
jest.mock('enquirer');

const runner = require('../lib/runner');
runner.mockImplementation(() => {});

describe('CLI Executer', () => {
    let cliExecuter = null;
    let multiCalls = 0;
    let multiChoices = null;
    let multiMapper = null;

    let inputCalls = 0;
    const inputConstructorObjs = [];

    beforeAll(() => {
        let inputRunCount = 0;

        const enquirer = require('enquirer');
        enquirer.MultiSelect = class MultiSelect {
            constructor(obj) {
                multiCalls++;
                multiChoices = obj.choices;
                multiMapper = obj.result;
            }

            run() {
                return ['--config', '--entry', '--progress'];
            }
        };
        enquirer.Input = class Input {
            constructor(obj) {
                this.mapper = obj.result;
                inputCalls++;
                inputConstructorObjs.push(obj);
            }

            run(obj) {
                inputRunCount++;
                return this.mapper(`test${inputRunCount}`);
            }
        };

        cliExecuter = require('../lib/utils/cli-executer');
    });

    it('runs enquirer options', async () => {
        await cliExecuter();
        expect(runner.mock.calls).toMatchSnapshot();
        expect(multiCalls).toEqual(1);
        expect(multiChoices).toMatchSnapshot();
        expect(typeof multiMapper).toEqual('function');
        expect(multiMapper(['--test1: test flag', '--test2: test flag 2'])).toEqual(['--test1', '--test2']);

        expect(inputCalls).toEqual(2);
        expect(inputConstructorObjs).toMatchSnapshot();
    });
});
