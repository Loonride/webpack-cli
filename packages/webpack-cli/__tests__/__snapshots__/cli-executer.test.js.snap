// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI Executer runs enquirer options 1`] = `
Array [
  Array [
    Array [],
    Array [
      "--config",
      "test1",
      "--entry",
      "test2",
      "--progress",
    ],
  ],
]
`;

exports[`CLI Executer runs enquirer options 2`] = `
Array [
  "--entry: The entry point of your application.",
  "--config: Provide path to a webpack configuration file",
  "--merge: Merge a configuration file using webpack-merge",
  "--progress: Print compilation progress during build",
  "--silent: Disable any output that webpack makes",
  "--help: Outputs list of supported flags",
  "--defaults: Allow webpack to set defaults aggresively",
  "--output: Output location of the file generated by webpack",
  "--plugin: Load a given plugin",
  "--global: Declares and exposes a global variable",
  "--target: Sets the build target",
  "--watch: Watch for files changes",
  "--hot: Enables Hot Module Replacement",
  "--sourcemap: Determine source maps to use",
  "--prefetch: Prefetch this request",
  "--json: Prints result as JSON",
  "--standard: Prints standard output",
  "--dev: Run development build",
  "--prod: Run production build",
  "--mode: Defines the mode to pass to webpack",
  "--version: Get current version",
  "--node-args: NodeJS flags",
]
`;

exports[`CLI Executer runs enquirer options 3`] = `
Array [
  Object {
    "initial": null,
    "message": "Enter value of the --config flag",
    "name": "value",
    "result": [Function],
  },
  Object {
    "initial": null,
    "message": "Enter value of the --entry flag",
    "name": "value",
    "result": [Function],
  },
]
`;
